frontend/catalyst/passes/builtin_passes.py

```
@@ -780,3 +782,109 @@ def circuit():
+
+        @qjit(pipelines=pipe, target="mlir")
+        def test_convert_clifford_to_ppr_workflow():
+
+            device = qml.device("lightning.qubit", wires=2)
+
+            @merge_ppr_ppm
+            @commute_ppr(max_pauli_size=2)
+            @to_ppr
+            @qml.qnode(device)
+            def f():
+                qml.CNOT([0, 2])
+                qml.T(0)
+                return measure(0), measure(1)
+
+            @merge_ppr_ppm(max_pauli_size=1)
+            @commute_ppr
+            @to_ppr
+            @qml.qnode(device)
+            def g():
+                qml.CNOT([0, 2])
+                qml.T(0)
+                qml.T(1)
+                qml.CNOT([0, 1])
+                return measure(0), measure(1)
+
+            return f(), g()
+
+        ppm_specs = get_ppm_specs(test_convert_clifford_to_ppr_workflow)
+        print(ppm_specs)
+
+    Example PPM Specs:
+
+    .. code-block:: pycon
+
+        . . .
+        {
+            'f_0': {
+                        'max_weight_pi8': 1,
+                        'num_logical_qubits': 2,
+                        'num_of_ppm': 2,
+                        'num_pi8_gates': 1
+                    },
+            'g_0': {
+                        'max_weight_pi4': 2,
+                        'max_weight_pi8': 1,
+                        'num_logical_qubits': 2,
+                        'num_of_ppm': 2,
+                        'num_pi4_gates': 3,
+                        'num_pi8_gates': 2
+                    }
+        }
+        . . .
+
+    """
+
+    if QJIT.mlir is not None:
+        # aot mode
+        raw_result = _quantum_opt(
+            ("--pass-pipeline", "builtin.module(ppm-specs)"), [], stdin=QJIT.mlir_opt
+        )
+        try:
+            return json.loads(
+                raw_result[: raw_result.index("module")]
+            )  # remove MLIR starting with substring "module..."
+        except Exception as e:
+            print(e)
+            raise
```

Let's raise an actual error with a message no? :sweat_smile: 
```suggestion
            raise CompileError(f"Invalid json format encountered in get_ppm_specs. Got {raw_result[: raw_result.index("module")]}")
```

(You will need to `from catalyst.utils.exceptions import CompileError`)